File: BubbleSort.java
import java.util.Random;

public class BubbleSort {

    public static void bubbleSort(int[] arr, int size) {

        for (int i = 0; i < size - 1; i++) {

            for (int j = 0; j < size - i - 1; j++) {

                if (arr[j] > arr[j + 1]) {

                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static int[] generateNumbers(int[] arr, int size) {

        Random random = new Random();

        for (int i = 0; i < size; i++) {

            arr[i] = random.nextInt(100000);
        }

        return arr;
    }

    public static void main(String[] args) {

        int size1 = 40000;
        int size2 = 80000;

        int[] arr1 = new int[size1];
        int[] arr2 = new int[size2];

        generateNumbers(arr1, size1);

        long start1 = System.nanoTime();

        bubbleSort(arr1, size1);

        long end1 = System.nanoTime();

        double time1 = (end1 - start1) / 1_000_000_000.0;

        generateNumbers(arr2, size2);

        long start2 = System.nanoTime();

        bubbleSort(arr2, size2);

        long end2 = System.nanoTime();

        double time2 = (end2 - start2) / 1_000_000_000.0;

        System.out.printf("Ratio : %.2f%n", time2 / time1);
    }
}

File: bubble_sort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void bubbleSort(int *arr, int size)
{
    for(int i = 0; i < size - 1; i++)
    {
        for(int j = 0; j < size - i - 1; j++)
        {
            if(arr[j] > arr[j + 1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                arr[j] = temp;
            }
        }
    }
}

int* generateNumbers(int *arr, int size)
{
    for(int i = 0; i < size; i++)
    {
        arr[i] = rand() % 100000;
    }

    return arr;
}


int main()
{
    srand(time(NULL));

    int size1 = 40000;
    int size2 = 80000;

    int *arr1 = (int*)malloc(size1 * sizeof(int));
    int *arr2 = (int*)malloc(size2 * sizeof(int));

    arr1 = generateNumbers(arr1, size1);

    clock_t start1 = clock();

    bubbleSort(arr1, size1);

    clock_t end1 = clock();

    double time1 =
    ((double)(end1 - start1)) / CLOCKS_PER_SEC;

    arr2 = generateNumbers(arr2, size2);

    clock_t start2 = clock();

    bubbleSort(arr2, size2);

    clock_t end2 = clock();

    double time2 =
    ((double)(end2 - start2)) / CLOCKS_PER_SEC;

    printf("Ratio : %.2f\n", time2 / time1);

    free(arr1);
    free(arr2);

    return 0;
}
